src/components/Patient/LinkDepartmentsSheet.tsx

```diff
@@ -219,9 +223,9 @@ export default function LinkDepartmentsSheet({
                       variant="ghost"
                       size="icon"
                       onClick={() => removeOrganization(org.id)}
-                      disabled={isRemoving}
+                      disabled={removingOrgId === org.id}
                     >
-                      {isRemoving ? (
+                      {removingOrgId === org.id ? (
                         <Loader2 className="h-4 w-4 animate-spin" />
                       ) : (
                         <Trash2 className="h-4 w-4 text-destructive" />
```

Comment:
i'd say wrap this button in a component and keep the useMutation there in that component, and use useMutation's `isPending` to show the loading state instead of creating a new state here;